File: generated-code/example/cpp/model/Structure.cpp
#include "Structure.hpp"

Structure::Structure() { }

Structure::Structure(std::string text, float floatNumber, double doubleNumber) : text(text), floatNumber(floatNumber), doubleNumber(doubleNumber) { }

// Read Structure from input stream
Structure Structure::readFrom(InputStream& stream) {
    std::string text;
    text = stream.readString();
    float floatNumber;
    floatNumber = stream.readFloat();
    double doubleNumber;
    doubleNumber = stream.readDouble();
    return Structure(text, floatNumber, doubleNumber);
}

// Write Structure to output stream
void Structure::writeTo(OutputStream& stream) const {
    stream.write(text);
    stream.write(floatNumber);
    stream.write(doubleNumber);
}

// Get string representation of Structure
std::string Structure::toString() const {
    std::stringstream ss;
    ss << "Structure { ";
    ss << "text: ";
    ss << '"' << text << '"';
    ss << ", ";
    ss << "floatNumber: ";
    ss << floatNumber;
    ss << ", ";
    ss << "doubleNumber: ";
    ss << doubleNumber;
    ss << " }";
    return ss.str();
}

File: generated-code/example/ruby/model/structure.rb
class Structure
    attr_accessor :text
    attr_accessor :float_number
    attr_accessor :double_number

    def initialize(text, float_number, double_number)
        @text = text
        @float_number = float_number
        @double_number = double_number
    end

    def self.read_from(stream)
        text = stream.read_string()
        float_number = stream.read_float()
        double_number = stream.read_double()
        Structure.new(text, float_number, double_number)
    end

    def write_to(stream)
        stream.write_string(@text)
        stream.write_float(@float_number)
        stream.write_double(@double_number)
    end

    def to_s
        string_result = "Structure { "
        string_result += "text: "
        string_result += @text.dump
        string_result += ", "
        string_result += "float_number: "
        string_result += @float_number.to_s
        string_result += ", "
        string_result += "double_number: "
        string_result += @double_number.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end